### ISSUE TITLE
Empty output, regardless of what I change

### ISSUE THREAD

**Issue by ryenots** [Ltamann/ComfyUI-TBG-SAM3#12]
I've been trying to get SAM3 functional, using this node pack, and a few others. I'm not what the issue is, but the output never produces any sort of result. Here are a few example workflows using your node pack. 

<img width="2145" height="1069" alt="Image" src="https://github.com/user-attachments/assets/79df2adc-7d1b-4172-8f0b-2a36d3a9ecde" />

<img width="2260" height="1089" alt="Image" src="https://github.com/user-attachments/assets/d877ee40-cb4d-4e7b-a54b-20bd3e56c5b8" />

This is on a fresh install of Comfy, with the only custom nodes installed being yours. Using python 3.12.12, torch 2.7, and cuda 12.6, as listed on https://github.com/facebookresearch/sam3 prerequisites. OS is Ubuntu. 

The only thing that is non-standard about my install, is that I've downloaded ```sam3.pt``` from https://www.modelscope.cn/models/facebook/sam3/files rather than huggingface, in order to avoid the gated access application. As far as I understand this shouldn't make a difference, though I could be wrong.

The following is a output for ComfyUI starting, and executing the prompt for the airplane image.
```
[ComfyUI-Manager] Using uv as Python module for pip operations.
Using Python 3.12.12 environment at: venv
[START] Security scan
[DONE] Security scan
## ComfyUI-Manager: installing dependencies done.
** ComfyUI startup time: 2025-11-24 11:34:33.559
** Platform: Linux
** Python version: 3.12.12 (main, Oct 10 2025, 08:52:57) [GCC 11.4.0]
** Python executable: /home/ryenots/apps/ComfyUI/venv/bin/python3
** ComfyUI Path: /home/ryenots/apps/ComfyUI
** ComfyUI Base Folder Path: /home/ryenots/apps/ComfyUI
** User directory: /home/ryenots/apps/ComfyUI/user
** ComfyUI-Manager config path: /home/ryenots/apps/ComfyUI/user/default/ComfyUI-Manager/config.ini
** Log path: /home/ryenots/apps/ComfyUI/user/comfyui.log
Using Python 3.12.12 environment at: venv
Using Python 3.12.12 environment at: venv

Prestartup times for custom nodes:
   0.7 seconds: /home/ryenots/apps/ComfyUI/custom_nodes/ComfyUI-Manager

Checkpoint files will always be loaded safely.
Total VRAM 24253 MB, total RAM 257586 MB
pytorch version: 2.9.1+cu128
Set vram state to: NORMAL_VRAM
Device: cuda:0 NVIDIA GeForce RTX 3090 : cudaMallocAsync
Enabled pinned memory 244706.0
working around nvidia conv3d memory bug.
Using pytorch attention
Python version: 3.12.12 (main, Oct 10 2025, 08:52:57) [GCC 11.4.0]
ComfyUI version: 0.3.71
ComfyUI frontend version: 1.30.6
[Prompt Server] web root: /home/ryenots/apps/ComfyUI/venv/lib/python3.12/site-packages/comfyui_frontend_package/static
Total VRAM 24253 MB, total RAM 257586 MB
pytorch version: 2.9.1+cu128
Set vram state to: NORMAL_VRAM
Device: cuda:0 NVIDIA GeForce RTX 3090 : cudaMallocAsync
Enabled pinned memory 244706.0
### Loading: ComfyUI-Manager (V3.37.1)
[ComfyUI-Manager] network_mode: public
### ComfyUI Version: v0.3.71-6-gf66183a5 | Released on '2025-11-23'

======================================================================
[SAM3] ComfyUI SAM3 Node - Loading (TBG)
======================================================================
[SAM3] SAM3 Python package detected.
[ComfyUI-Manager] default cache updated: https://raw.githubusercontent.com/ltdrdata/ComfyUI-Manager/main/alter-list.json
[ComfyUI-Manager] default cache updated: https://raw.githubusercontent.com/ltdrdata/ComfyUI-Manager/main/model-list.json
[ComfyUI-Manager] default cache updated: https://raw.githubusercontent.com/ltdrdata/ComfyUI-Manager/main/github-stats.json
[ComfyUI-Manager] default cache updated: https://raw.githubusercontent.com/ltdrdata/ComfyUI-Manager/main/custom-node-list.json
[ComfyUI-Manager] default cache updated: https://raw.githubusercontent.com/ltdrdata/ComfyUI-Manager/main/extension-node-map.json
[SAM3] Registered 4 nodes:
[SAM3]  - TBG SAM3 Model Loader (TBGLoadSAM3Model)
[SAM3]  - TBG SAM3 Model Loader and Downloader (TBGSAM3ModelLoaderAdvanced)
[SAM3]  - TBG SAM3 Segmentation (TBGSam3Segmentation)
[SAM3]  - TBG SAM3 Selector (TBGSAM3PromptCollector)
======================================================================

Import times for custom nodes:
   0.0 seconds: /home/ryenots/apps/ComfyUI/custom_nodes/websocket_image_save.py
   0.0 seconds: /home/ryenots/apps/ComfyUI/custom_nodes/execution-inversion-demo-comfyui
   0.1 seconds: /home/ryenots/apps/ComfyUI/custom_nodes/ComfyUI-Manager
   0.5 seconds: /home/ryenots/apps/ComfyUI/custom_nodes/tbg-sam3

Context impl SQLiteImpl.
Will assume non-transactional DDL.
No target revision found.
Starting server

To see the GUI go to: http://127.0.0.1:8188
FETCH ComfyRegistry Data: 5/108
FETCH ComfyRegistry Data: 10/108
FETCH ComfyRegistry Data: 15/108
FETCH ComfyRegistry Data: 20/108
got prompt
[TBGSAM3ModelLoaderAdvanced] Using selected local checkpoint: /home/ryenots/apps/ComfyUI/models/sam3/sam3.pt
FETCH ComfyRegistry Data: 25/108
FETCH ComfyRegistry Data: 30/108
FETCH ComfyRegistry Data: 35/108
FETCH ComfyRegistry Data: 40/108
[TBGSAM3ModelLoaderAdvanced] SAM3 model ready on device: cuda
[SAM3] pipeline_mode='all', instances=True | pos_boxes_active=False, neg_boxes_active=False, pos_points_active=False, neg_points_active=False
[SAM3] Moving model from cuda:0 to cuda
[SAM3] Running segmentation
[SAM3] Confidence threshold: 0.1
[SAM3] Image size: (393, 280)
[SAM3] Using text_prompt='airplane'
[SAM3 PROCESSOR DEBUG] Raw predictions from model:
  pred_logits shape: torch.Size([1, 200, 1])
  pred_boxes shape: torch.Size([1, 200, 4])
  pred_masks shape: torch.Size([1, 200, 288, 288])
  out_probs (after sigmoid) range: [nan, nan]
  presence_score: nan
  out_probs (after presence multiply) range: [nan, nan]
  out_probs shape: torch.Size([1, 200])
  Threshold: 0.1
  Predictions passing threshold: 0 / 1
[SAM3] total box prompts=0
[SAM3] total point prompts=0
[SAM3 DEBUG] RAW PREDICTIONS: total 0
[SAM3 DEBUG] Output boxes shape: torch.Size([0, 4])
[SAM3] No detections found at threshold 0.1
Prompt executed in 11.50 seconds
```

Finally, I apologize if this is not the best place for this question, as I doubt this is because of your nodes, since other similar node packs for SAM3 give me the same issue. Loosing my sanity trying to solve this, so I though I may as well reach out to see if I am missing something. Thank you

**Comment by Ltamann** [comment]
I’ve just did the node — the point and mask selectors are working correctly. Try adjusting your prompts, check what the meta suggests, and test your prompts on the official website. Prompts are a Model problem not node related 

<img width="3012" height="1336" alt="Image" src="https://github.com/user-attachments/assets/d4813329-f378-4e70-b3b1-a0c8967b778c" />

**Comment by ryenots** [comment]
Hi, sorry for the delayed reply. I've replicated your workflow, and things are working now. I think my issue was most likely the min_size parameter. Thank you for the help!
